hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -67,21 +77,84 @@ public final class OmSnapshotManager implements AutoCloseable {
   private final OzoneManager ozoneManager;
   private final SnapshotDiffManager snapshotDiffManager;
   private final LoadingCache<String, OmSnapshot> snapshotCache;
-  private final ManagedRocksDB snapshotDiffDb;
+  private ManagedRocksDB snapshotDiffDb;
+
+  /**
+   * Contains all the snap diff job which are either queued, in_progress or
+   * done. This table is used to make sure that there is only single job for
+   * similar type of request at any point of time.
+   * |----------------------------------------------|
+   * |  KEY                         |  VALUE        |
+   * |----------------------------------------------|
+   * |  fromSnapshotId-toSnapshotId | snapDiffJodId |
+   * |----------------------------------------------|
+   */
+  private static final String SNAP_DIFF_JOB_TABLE_NAME =
+      "snap-diff-job-table";
+
+  /**
+   * Global table to keep the diff report. Each key is prefixed by the jobId
+   * to improve look up and clean up. JodId comes from snap-diff-job-table.
+   * |--------------------------------|
+   * |  KEY         |  VALUE          |
+   * |--------------------------------|
+   * |  jodId-index | DiffReportEntry |
+   * |--------------------------------|
+   */
+  private static final String SNAP_DIFF_REPORT_TABLE_NAME =
+      "snap-diff-report-table";
+
+  private final ManagedColumnFamilyOptions columnFamilyOptions;
+  private final ManagedDBOptions options;
+
+  // TODO: [SNAPSHOT] create config for max allowed page size.
+  private final int maxPageSize = 1000;
 
   OmSnapshotManager(OzoneManager ozoneManager) {
-    this.ozoneManager = ozoneManager;
+    this.options = new ManagedDBOptions();
+    this.options.setCreateIfMissing(true);
+    this.columnFamilyOptions = new ManagedColumnFamilyOptions();
+
+    List<ColumnFamilyDescriptor> columnFamilyDescriptors = new ArrayList<>();
+    List<ColumnFamilyHandle> columnFamilyHandles = new ArrayList<>();
+    ColumnFamilyHandle snapDiffJobCf;
+    ColumnFamilyHandle snapDiffReportCf;
+    String dbPath = getDbPath(ozoneManager.getConfiguration());
+
+    try {
+      // Add default CF
+      columnFamilyDescriptors.add(new ColumnFamilyDescriptor(
+          StringUtils.string2Bytes(DEFAULT_COLUMN_FAMILY_NAME),
+          columnFamilyOptions));
```

Comment:
Do we want to put `SNAP_DIFF_JOB_TABLE_NAME` and `SNAP_DIFF_REPORT_TABLE_NAME` in the initial CF list as well?

Like here: https://github.com/apache/ozone/blob/1f8795b4fd9108865f8b1f52bbe3de48ba5f324a/hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java#L590-L598

Reply:
Ok sounds good